tests/unit/cmd/hmsl/test_check.py

```diff
@@ -0,0 +1,70 @@
+from uuid import uuid4
+
+import pytest
+from click.testing import CliRunner
+
+from ggshield.cmd.main import cli
+from tests.unit.conftest import assert_invoke_ok, my_vcr
+
+
+@pytest.fixture(autouse=True)
+def use_staging_env(monkeypatch):
+    # The cassettes where recorded before the service was operational in production
+    monkeypatch.setenv(
+        "GITGUARDIAN_HMSL_URL", "https://hasmysecretleaked.staging.gitguardian.tech"
+    )
+
+
+@my_vcr.use_cassette
+def test_hmsl_check_random_secret(cli_fs_runner: CliRunner) -> None:
+    """
+    GIVEN a random
+    WHEN running the check command on it
+    THEN we should not find a match
+    """
+    non_matching_secret = str(uuid4())
+    with open("secrets.txt", "w") as f:
+        f.write(non_matching_secret + "\n")
+
+    result = cli_fs_runner.invoke(cli, ["hmsl", "check", "secrets.txt"])
+
+    assert "Found 0 leaked secrets" in result.output
+    assert_invoke_ok(result)
```

Comment:
It's better to call `assert_invoke_ok()` before the other asserts because in case of failure, it prints all of stdout and stderr.